examples/kuzu/Papers-100M/train.py

```diff
@@ -0,0 +1,150 @@
+# Please refer to: https://kuzudb.com/docs/client-apis/python-api/overview.html
+# for how to use the Python API of Kùzu.
+########################################################################
+# The buffer pool size is set to 40GB. You can change it to a smaller value
+# if you have less memory.
+KUZU_BM_SIZE = 40 * 1024**3
+# Create kuzu database
+db = kuzu.Database("papers100M", KUZU_BM_SIZE)
+# Get remote backend for torch geometric
+feature_store, graph_store = db.get_torch_geometric_remote_backend(
+    mp.cpu_count())
+
+# Plug the graph store and feature store into the NeighborLoader
+loader_kuzu = NeighborLoader(
+    data=(feature_store, graph_store), num_neighbors={
+        ('paper', 'cites', 'paper'): [30, 30]
+    }, batch_size=LOADER_BATCH_SIZE, input_nodes=('paper', input_nodes),
+    input_time=None, num_workers=4)
+
+
+# Define the model for training. The model is ported from
+# https://github.com/mengyangniu/ogbn-papers100m-sage
+class MLP(nn.Module):
+    def __init__(self, input_dim, hidden_dim, embed_dim, num_layers: int,
+                 act: str = 'ReLU', bn: bool = False, end_up_with_fc=False,
+                 bias=True):
+        super(MLP, self).__init__()
+        self.module_list = []
+        for i in range(num_layers):
+            d_in = input_dim if i == 0 else hidden_dim
+            d_out = embed_dim if i == num_layers - 1 else hidden_dim
+            self.module_list.append(nn.Linear(d_in, d_out, bias=bias))
+            if end_up_with_fc:
+                continue
+            if bn:
+                self.module_list.append(nn.BatchNorm1d(d_out))
+            self.module_list.append(getattr(nn, act)(True))
+        self.module_list = nn.Sequential(*self.module_list)
+
+    def forward(self, x):
+        return self.module_list(x)
+
+
+class SAGE(nn.Module):
+    def __init__(self, in_feats, n_hidden, n_classes, n_layers, activation,
+                 dropout):
+        super().__init__()
+        self.n_layers = n_layers
+        self.n_hidden = n_hidden
+        self.n_classes = n_classes
+        self.layers = nn.ModuleList()
+        self.bns = nn.ModuleList()
+        self.layers.append(
+            SAGEConv(in_feats, n_hidden, aggr='mean', bias=False))
+        self.bns.append(torch.nn.BatchNorm1d(n_hidden))
+        for i in range(1, n_layers - 1):
+            self.layers.append(
+                SAGEConv(n_hidden, n_hidden, aggr='mean', bias=False))
+            self.bns.append(torch.nn.BatchNorm1d(n_hidden))
+        self.layers.append(
+            SAGEConv(n_hidden, n_hidden, aggr='mean', bias=False))
+        self.bns.append(torch.nn.BatchNorm1d(n_hidden))
+        self.mlp = MLP(in_feats + n_hidden * n_layers, 2 * n_classes,
+                       n_classes, num_layers=2, bn=True, end_up_with_fc=True,
+                       act='LeakyReLU')
+        self.dropout = nn.Dropout(dropout)
+        self.activation = activation
+        self.profile = locals()
```

Comment:
Can you clarify why we need this?

Reply:
It was ported from the DGL example and I was not sure if/why this was needed. Since now I switched to use `torch_geometric.nn.models.MLP`, this is removed.